### ISSUE TITLE
I can't generate a presigned url for certain buckets.

### ISSUE THREAD

**Issue by jumbalaya-wanton** [aws/aws-sdk-ruby#444]
I'm not able to generate a pre-signed for a bucket with dots in the name: `c.foo.com`, for example. The request is getting cancelled by the browser, and I suspect it has to do with using https.

![request1](https://f.cloud.github.com/assets/6363612/1882529/481b3ba4-7977-11e3-9d22-b94d2287e4ad.jpg)
![request2](https://f.cloud.github.com/assets/6363612/1882530/481ca37c-7977-11e3-8efa-832816122305.jpg)

Notice below what happens when I try to access the bucket using the browser.

![request3](https://f.cloud.github.com/assets/6363612/1882531/48207092-7977-11e3-8f5e-8121c6dc4ac9.jpg)
![request4](https://f.cloud.github.com/assets/6363612/1882532/48215688-7977-11e3-8128-c0f678013e6f.jpg)

Basically, the request for a presigned url is getting cancelled because it's getting routed to `https://c.foo.co.s3.amazonaws.com`. It's running into the same issue as if I were trying to access the url in the browser.

How do I solve this?

Here's a question related to this issue: http://stackoverflow.com/questions/21031341/create-a-presigned-url-for-s3-bucket-with-a-vanity-domain
